pkg/cgroups/cgroupsv2/runtime.go

```diff
@@ -0,0 +1,147 @@
+package cgroupsv2
+
+import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"log/slog"
+	"os"
+	"os/exec"
+	"syscall"
+	"time"
+
+	"github.com/google/uuid"
+	jobv1 "github.com/kralicky/jobserver/pkg/apis/job/v1"
+	"github.com/kralicky/jobserver/pkg/cgroups"
+	"github.com/kralicky/jobserver/pkg/jobs"
+	"github.com/kralicky/jobserver/pkg/util"
+)
+
+const gracePeriod = 10 * time.Second
+
+type v2Runtime struct {
+	mgr *cgroupManager
+}
+
+func NewRuntime() (jobs.Runtime, error) {
+	mgr, err := newCgroupManager()
+	if err != nil {
+		return nil, fmt.Errorf("failed to setup jobserver cgroup: %w", err)
+	}
+
+	return &v2Runtime{
+		mgr: mgr,
+	}, nil
+}
+
+// Execute implements jobs.Runtime.
+func (l *v2Runtime) Execute(ctx context.Context, spec *jobv1.JobSpec) (jobs.Process, error) {
+	cmdSpec := spec.GetCommand()
+
+	// generate a uuid for the process, but encode it in the raw hex format.
+	// this makes it slightly easier to use from the command line, since many
+	// terminals treat '-' as a word separator, making it difficult to copy
+	// the whole string using a double-click.
+	u := uuid.New()
+	id := hex.EncodeToString(u[:])
+
+	cmd := exec.CommandContext(ctx, cmdSpec.GetCommand(), cmdSpec.GetArgs()...)
+	cmd.Env = append(os.Environ(), cmdSpec.GetEnv()...)
```

Comment:
Does it safe to pass all os.Env to the user process ?

Reply:
Probably not :smile:. Since handling environment variables (and working directory, etc) is out of spec, I assume we're mostly ignoring this, and inheriting the server's environment would be needed as a workaround.